README.md
# JETRIS - A Java-Based Tetris Clone

[![Build Status](https://travis-ci.org/devng/jetris.svg?branch=master)](https://travis-ci.org/devng/jetris)

## TL;DR

JETRIS is a Java-based Tetris clone for the desktop. JETRIS is focused on a fast gaming, so a game usually takes about 10 min.

![](screenshot.png?raw=true)

### This Readme includes:

1. System Requirements
2. How to Build JETRIS
  * How to build an Windows EXE
  * How to build a Mac OS APP and DMG
  * How to extend JETRIS
3. Running JETRIS on Your System
4. How to Play
5. Scoring System
6. Saving Your Old HiScores

1. System Requirements
----------------------

JETRIS is written in the JAVA programming language, this means that it can be run on any Operating System which has JAVA Runtime Environment (JRE).

You need JRE 1.7.0 or above. You can download JRE for free at www.java.com

2. How to Build JETRIS
----------------------
If you want to build JETRIS on your local machine you need to install the Java Development Kit (JDK). You need JDK 1.7.0 or above.

JETRIS uses the Gradle build system, you can build the project locally just by typing the following in the console:

```
./gradlew build
```

On Windows use:
```
./gradlew.bat build
```

The generated jar file can be found in the folder _build/libs_.

### How to build an Windows EXE
The Gradle build script provides a way to build an Windows executable. You can even do this on a non-Windows machine. For that purpose you need an external application called
[Launch4j](http://launch4j.sourceforge.net/). Download it and install it on your local machine. The version used to test this feature is 3.8, but it will probably work with
other versions too. In addition to this, you need to create an environment variable called `LAUNCH4J_HOME` which points to the folder where you have installed the application.

After that you can build the exe file by running:
```
./gradlew launch4j
```

The generated exe file can be found in the folder _build/launch4j_.

### How to build a Mac OS APP and DMG
The JETRIS Gradle build script uses [MacAppBundle Plugin](https://github.com/crotwell/gradle-macappbundle) to generate Mac OS X app packages _and_ a _dmg_ distribution file.
This is done automatically on Mac OS X with each Gradle build.

The generated app package can be found in the folder _build/macApp_ and the dmg file in _build/distributions_.

### How to extend JETRIS
If you want to import the project in an IDE such as Eclipse or IntelliJ IDEA then Gradle provides a way to generate all the necessary project files.

Generate Eclipse project:
```
./gradlew eclipse
```

Generate IntelliJ IDEA project:
```
./gradlew idea
```

3. Running JETRIS on Your System
--------------------------------

To start JETRIS try one of the following options: 

* Double click on the JAR File to start JETRIS. If this didn't work, then you didn't associate your JAR Files with your JRE.

* For Windows users we also provide an exe file, which ist just a wrapper for the JAR file. Double click on the exe file to start the application.

* Double click on `JETRIS.bat` for Windows users or on `JETRIS` for Linux users. This will start the application, but only if you have built it with Gradle first.

* Open the console go to your JETRIS folder and type: 
```
java -jar jetris-1.2.jar
```

4. How to Play
--------------

Use the following keys to play JETRIS:

* A or Left Arrow - Move the figure to left
* D or Right Arrow - Move the figure to right
* W or Up Arrow - Rotate the figure
* S or Down Arrow - Move the figure quick down
* Space - Drop the figure immediately
* P or 'Pause' Button - Pause
* R or 'Restart' Button - Restart
* H - View HiScore
* Esc - Exit

5. Scoring System
-----------------

* Clearing 1 Line at once, gives You 100 points + 5 x the current level
* Clearing 2 Line at once, gives You 400 points + 20 x the current level
* Clearing 3 Line at once, gives You 900 points + 45 x the current level
* Clearing 4 Line at once, gives You 1600 points + 80 x the current level

### For example: 

The current level is 20 (the highest level) and You clear 4 Lines at once, then You get 1600 + 80 x 20 = 2 x 1600 = 3200. So on level 20 you are making twice as much points as on level 0.

6. Saving Your Old HiScores
---------------------------

Copy the old `JETRIS.DAT` File to your new version of JETRIS folder.


CHANGES.md
2015-08-07
----------
* JETRIS Version 1.2
* Add Launch4J configuration and a new Gradle taks launch4j for generating an Windows exe file
* Use a Gradle plugin to create MAC OS X app package and a dmg file
* Remove the publish HiScores functionality, as it did not work
* Remove the webapp source folder, since there is no web page anymore
* Small code improvements
* Update Gradle wrapper to version 2.5
* Update the documentation

2015-08-05
----------
* JETRIS Version 1.1.1
* Use Gradle as a build system
* Use the standard directory layout
* Change the package name to com.devng.jetris
* Adjust the documentation
* Include the web page files in the repo
* Small code adjustments

2015-08-03
----------
* Move the project from sourceforge.net to GitHub
* Change the licence from MPL 1.1 to MPL 2.0
* Change documentation format from TXT to MD

2006-09-28
----------
* JETRIS version 1.1
* Help dialog in the help menu added
* Splash screen added
* Faster HiScore publishing
* HiScore are now painted red, so a player wont get confused
* Fixed bug when pressing the cancel button on 'Enter your name' dialog
* A pause bug was fixed

2006-09-26
----------
* Initial release, JETRIS version 1.0


src/main/webapp/scoringsystem.php
<?php
include "templateUp.php";
?>
		<h1>Scoring System</h1>
		<p>All of this information you can also find in the ReadMe.txt file of Your JETRIS.</p>
		<br>
		<ul>
			<li>Clearing 1 Line at once, gives You 100 points + 5 x the current level</li>
			<li>Clearing 2 Line at once, gives You 400 points + 20 x the current level</li>
			<li>Clearing 3 Line at once, gives You 900 points + 45 x the current level</li>
			<li>Clearing 4 Line at once, gives You 1600 points + 80 x the current level</li>
		</ul>
		<br>
		<div class="rule"></div>
		
		<h2>For example:</h2>
		<p>The current level is 20 (the highest level) and You clear 4 Lines at once, then You get 1600 + 80 x 20 = 2 x 1600 = 3200. on level 20 you are making twice as much points as on level 0.</p>
		<div class="rule"></div>
<?php
include "templateDown.php";
?>